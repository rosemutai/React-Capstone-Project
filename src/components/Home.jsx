import { useSelector } from 'react-redux';
import CountryCard from './CountryCard';

const Home = () => {
  const statistics = useSelector((state) => state.statistics.statistics);
  console.log(statistics);

  return (
    <div className="w-full overflow-x-hidden">
      <h3 className="text-center text-sm font-semibold tracking-wide mt-4">
        Covid19 Statistics
      </h3>
      <div className="w-full my-8">
        <form className="search w-10/12 mx-auto flex gap-3">
          <input
            type="text"
            name="search"
            placeholder="Search by country"
            className="p-2 rounded-md w-full
            border border-slate-200"
          />
          <button type="submit">
            <span className="material-symbols-outlined text-slate-200 ">
              search
            </span>
          </button>
        </form>
      </div>
      <h5 className="font-bold tracking-wider text-sm px-2 text-white absolute z-50">STATS BY COUNTRY</h5>
      <div className="largest-population w-full grid grid-cols-2 px-2 mt-20">
        {statistics &&
          statistics.map((item) => (
            <CountryCard
              key={item.country}
              country={item.country}
              active={item.active}
            />
          ))}
      </div>
    </div>
  );
};

export default Home;
