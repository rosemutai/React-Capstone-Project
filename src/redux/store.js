/* eslint-disable import/prefer-default-export */
import { configureStore } from '@reduxjs/toolkit';
import StatisticsReducer from './statistics/statisticsSlice';

export const store = configureStore({
  reducer: {
    statistics: StatisticsReducer,
  },
});
