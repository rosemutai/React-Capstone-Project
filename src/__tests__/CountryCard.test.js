import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import { store } from '../redux/store';

it('renders CardComponent correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>

      </BrowserRouter>,

    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
