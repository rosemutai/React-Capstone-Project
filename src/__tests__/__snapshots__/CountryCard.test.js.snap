// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardComponent correctly 1`] = `
<div
  className="w-full overflow-x-hidden"
>
  <h3
    className="text-center text-lg font-semibold text-white tracking-wider mt-4"
  >
    Covid19 Statistics
  </h3>
  <div
    className="w-full my-8"
  >
    <form
      className="search w-10/12 mx-auto flex gap-3"
      onSubmit={[Function]}
    >
      <input
        className="p-2 rounded-md w-full placeholder:text-gray-300 placeholder:text-sm border border-transparent focus:outline-0 focus:shadow-inner focus:shadow-hotPink"
        name="search"
        onChange={[Function]}
        placeholder="Search by country"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        <span
          className="material-symbols-outlined text-hotPink/60 p-2 rounded-md bg-white "
        >
          search
        </span>
      </button>
    </form>
  </div>
  <h5
    className="font-bold tracking-wider text-xs px-2 text-white absolute z-50"
  >
    COVID19 ACTIVE CASES BY COUNTRY
  </h5>
  <div
    className="largest-population w-full grid grid-cols-2 px-2 mt-20"
  />
</div>
`;
